### DISCUSSION TITLE
Custom headers

### DISCUSSION THREAD

**Discussion by dpetka2001** [ibhagwan/fzf-lua#1301] [Q&A] [answered]
I'm interested in creating some custom header for some actions. I saw this [comment](https://github.com/ibhagwan/fzf-lua/issues/946#issuecomment-1848454009) from the maintainer himself and indeed tried to pass `--header` in `fzf_opts` and it worked like a charm. However, the custom header is all orange compared to `fzf-lua`'s headers for builtin pickers, which use different color for `header_bind` and `header_text`. 

Is it possible to pass highlight groups to the custom `fzf_opts["--header"]` so it looks like `fzf-lua` default headers?

**Comment by dpetka2001** [comment, marked as answer]
I managed to do it by using `string.format` to format the header and `utils.ansi_from_hl` to color each string token.

**Reply by ibhagwan** [reply]
This is correct, fzf is run in the terminal and as such you need to convert neovim highlights to their corresponding colors using ansi escape sequences (fg, bg, bold, etc), using `utils.ansi_from_hl` is the easiest way as fzf-lua will do that for you, btw, setting headers has a shortcut (like prompt) meaning you can set `header=…` directly instead of using fzf_opts, for example `:FzfLua files header=foo` or `:lua require'fzf-lua'.files({ header = … })`.

**Reply by dpetka2001** [reply]
Oh thank you. I wasn't aware of that. However, I'm setting up some specific actions for a custom picker (the idea was taken from another user here where he created a project picker), so I have some actions for that picker
```lua
    actions = {
      ["default"] = function(selected)
        fzf.git_files({ cwd = selected[1] })
      end,
      ["ctrl-f"] = function(selected)
        fzf.grep_project({ cwd = selected[1] })
      end,
      ["ctrl-r"] = function(selected)
        fzf.oldfiles({ cwd = selected[1] })
      end,
    },
```
and I want to set some custom headers for these actions. Since all that is in a `util.fzf` module and I just call it from a keymap I believe it's easier (or more comfortable for me at least) to define that directly in `fzf_opts["--header"]`. But thank you for your suggestion :smile:

**Reply by dpetka2001** [reply]
Just a question @ibhagwan. The `fzf_opts` that I set up for the custom function will they override the general `fzf_opts`? Or will it be used just for the custom function? I mean the other pickers will still use the general `fzf_opts` and not the ones that are being defined in the custom function, correct?

PS: yes, just tested it. `FzfLua files` still has the default `fzf_opts`.

**Reply by ibhagwan** [reply]
Each action will use its own opts, in your case you’re only sending `cwd` so the rest will be the default opts, FYI, the original opts are also propagated in case you want to reuse:
```lua
      ["default"] = function(selected, opts)
        opts.cwd = selected[1]
        fzf.git_files(opts)
      end,
```

**Reply by dpetka2001** [reply]
Oh I see. Really appreciate all your help!! Grateful!! :smile:
